src/Features/CSharp/Portable/UsePatternMatching/CSharpAsAndNullCheckDiagnosticAnalyzer.cs

```diff
@@ -73,17 +74,42 @@ private void SyntaxNodeAction(SyntaxNodeAnalysisContext syntaxContext)
 
             var targetStatement = (StatementSyntax)node;
             var leftmostCondition = GetLeftmostCondition(targetStatement);
-            if (!leftmostCondition.IsKind(SyntaxKind.NotEqualsExpression, out BinaryExpressionSyntax notEquals))
+            if (leftmostCondition == null)
             {
                 return;
             }
 
-            var operand = GetNullCheckOperand(notEquals.Left, notEquals.Right)?.WalkDownParentheses();
+            bool isNegativeNullCheck;
+            switch (leftmostCondition.Kind())
+            {
+                case SyntaxKind.NotEqualsExpression:
+                    isNegativeNullCheck = false;
+                    break;
+                case SyntaxKind.EqualsExpression:
+                    isNegativeNullCheck = true;
+                    break;
+                default:
+                    return;
+            }
+
+            var comparison = (BinaryExpressionSyntax)leftmostCondition;
+            var operand = GetNullCheckOperand(comparison.Left, comparison.Right)?.WalkDownParentheses();
             if (operand == null)
             {
                 return;
             }
 
+            var semanticModel = syntaxContext.SemanticModel;
+            if (operand.IsKind(SyntaxKind.CastExpression, out CastExpressionSyntax castExpression))
```

Comment:
❓ Is the ability to handle a cast to `object` new for both the normal and negated forms? If so, we should add a test for the newly supported normal form.

Reply:
Yes, it's new, I'll add the test for `(object)x != null` as well.